### PR TITLE
Helper scripts for regression testing of cortx-k8s

### PR DESCRIPTION
**Pull request by walterlopatka**
These scripts are scripts that help me automate some of my personal regression testing for cortx-k8s.  These are not appropriate to put into the cortx-k8s repo because they are reliant on Seagate internal servers.  But it is important to put these somewhere accessible by others.

trigger_jenkins.py:  Script to trigger a CICD job that does a basic CORTX deploy + IO
post_jenkins_result.py:  Script to post to Seagate Jira (for evidence of test)

### CODE REVIEW

**Comment by walterlopatka** [comment]
The ssl "verify=False" issues are necessary due to Jenkins environment.  (I have chosen "ignore issue" in Codacy for future.)  The remaining docstring issue is a false alert... it has been addressed in the code.

File: K8s-Orch-Tools/jira_common.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    auth = userauth()
+    headers = {
+       "Accept": "application/json",
+       "Content-Type": "application/json",
+    }
+
+    retry_count = 0
+    while True:
+        try:
+            print("%s GET %s" % (time.ctime(), url), file=sys.stderr)
```
Comment (keithpine) [inline K8s-Orch-Tools/jira_common.py:44]:
```suggestion
            print(f"{time.ctime()} GET {url}", file=sys.stderr)
```
`%s` strings are so Python2.

File: K8s-Orch-Tools/jira_common.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    if r.status_code != 200:
+        raise FailedJqlException(r.text)
+
+    return json.loads(r.text)
+
+def query_all(jql, querydict=None, verbose=True):
+    # Take care of pagination of all queries.  Return a list of issues.
+
+    result = []
+    startAt = 0
```
Comment (keithpine) [inline K8s-Orch-Tools/jira_common.py:60]:
```suggestion
    start_at = 0
```
Prefer snake case.

File: K8s-Orch-Tools/jira_common.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    maxResults = 100
+    while True:
+        if not querydict:
+            querydict = {}
+        querydict.update({'startAt': startAt, 'maxResults': maxResults})
+        qout = query(jql, querydict=querydict)
+        f = open("qout.debug", "w")
+        f.write(json.dumps(qout))
+        f.close()
```
Comment (keithpine) [inline K8s-Orch-Tools/jira_common.py:69]:
```suggestion
        with open("qout.debug", "w") as f:
            json.dump(qout, f)
```

File: K8s-Orch-Tools/post_jenkins_result.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    file_content = open(filesummary).read()
+    file1, file2 = file_content.split('\n\n')
+
+    content = '*Code tested*\n\n'
+    content += file1 + '\n'
+    content += '{code}\n'
+    content += file2 + '\n'
+    content += '{code}\n'
+    content += f'\n[See attached logs|{attachment_url}]\n'
```
Comment (keithpine) [inline K8s-Orch-Tools/post_jenkins_result.py:67]:
```suggestion
    content = f'''*Code tested*

{file1}
{{code}}
{file2}
{{code}}

[See attached logs|{attachment_url}]
'''
```
```

File: K8s-Orch-Tools/trigger_jenkins.py [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+
+
+import argparse
+import datetime
+import re
+import os
+import sys
+import time
+
+from yaml import load, Loader
```
Comment (keithpine) [inline K8s-Orch-Tools/trigger_jenkins.py:65]:
```suggestion
```
Importing from top level makes it harder to read, IMO. `yaml.load()` is better than `load()`.

File: K8s-Orch-Tools/trigger_jenkins.py [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+        sys.exit(1)
+
+#
+# Read hostnames from solution.yaml
+#
+if not os.path.exists(args.solution_file):
+    print("File {args.solution_file} does not exist.", file=sys.stderr)
+    sys.exit(1)
+
+solution_content = load(open(args.solution_file), Loader=Loader)
```
Comment (keithpine) [inline K8s-Orch-Tools/trigger_jenkins.py:116]:
```suggestion
solution_content = yaml.safe_load(open(args.solution_file))
```
Is it necessary to use `load()`? I think the solution file is fine with `safe_load()`, then you also don't need `Loader`.

File: K8s-Orch-Tools/trigger_jenkins.py [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+nodes = solution_content['solution']['nodes']
+hosts = [f"hostname={n['name']},user={args.node_user},password={args.node_password}" for n in nodes.values()]
+
+jenkins_build_url = JENKINS_JOB+'/buildWithParameters'
+
+hosts0 = hosts[0]
```
Comment (keithpine) [inline K8s-Orch-Tools/trigger_jenkins.py:122]:
```suggestion
```
Variable is unused.

File: K8s-Orch-Tools/trigger_jenkins.py [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+
+if not os.path.isdir(artifactdir):
+    os.mkdir(artifactdir)
+
+for artifact in builddata['artifacts']:
+    print(f"Downloading artifact: {artifactdir}/{artifact['fileName']}")
+    r = requests.get(build_url+f"artifact/{artifact['relativePath']}", verify=False)   # NOSEC
+    f = open(os.path.join(artifactdir, artifact['fileName']), 'wb')
+    f.write(r.content)
+    f.close()
```
Comment (keithpine) [inline K8s-Orch-Tools/trigger_jenkins.py:230]:
```suggestion
    with open(os.path.join(artifactdir, artifact['fileName']), 'wb') as f:
        f.write(r.content)
```

**Review by ujjwalpl — APPROVED** [review]

**Review by keithpine — APPROVED** [review]
Some random suggestions.

**Comment by hessio** [comment]
Hi @walterlopatka what is the status on this PR? can it be merged?

**Comment by shailesh-vaidya** [comment]
Closing as an obsolete

**Pull request closed without merging**